src/runtime_src/core/pcie/driver/linux/xocl/mgmtpf/mgmt-utils.c

```diff
@@ -383,10 +377,16 @@ long xclmgmt_hot_reset(struct xclmgmt_dev *lro, bool force)
 	xocl_clear_pci_errors(lro);
 	store_pcie_link_info(lro);
 
+	/* Clear previous state of device status */
+	lro->ready = true;
```

Comment:
ready is set unconditionally? I thought it relies on check_device_ready()?

Reply:
I removed that check as we did not want the driver to dictate device condition to the tool. All of the changes regarding the ready state have been reverted to their original state, with the exception of having the ready flag control the driver state in `xclmgmt_connect_notify`

Reply:
Actually, I do not think you can set ready unconditionally here, because you do not know that if the device is in ready state before issuing Hot reset.